src/include/gurt/types.h

```diff
@@ -32,15 +32,46 @@
  */
 
 #if defined(__cplusplus)
+#include <type_traits>
 extern "C" {
+#define d_check_type(var, type)	std::is_same<decltype(var), type>::value
```

Comment:
(style)  spaces required around that '<' (ctx:VxV)
(style)  spaces required around that '>' (ctx:VxO)
(style)  Macros with complex values should be enclosed in parentheses